src/hooks/useAuthUser.ts

```diff
@@ -1,25 +1,30 @@
-/*
-This hook wraps the useUser hook from @thirdweb-dev/react-core and adds our jwt-specific metadata we pass back to the client
-see https://portal.thirdweb.com/wallets/auth/server-frameworks/next#enhancing-session-data
-and https://github.com/Stand-With-Crypto/swc-web/blob/a99648eb4097dffb335155375b7d5b439c9b997c/src/utils/server/thirdweb/thirdwebAuthConfig.ts#L50
-*/
-
-import { Json } from '@thirdweb-dev/auth'
-import { useUser } from '@thirdweb-dev/react'
+import { useMemo } from 'react'
+import Cookies from 'js-cookie'
+import { jwtDecode } from 'jwt-decode'
+import { useActiveWalletConnectionStatus } from 'thirdweb/react'
 
 import { parseThirdwebAddress } from '@/hooks/useThirdwebAddress/parseThirdwebAddress'
-import { AuthSessionMetadata } from '@/utils/server/thirdweb/types'
+import { THIRDWEB_AUTH_TOKEN_COOKIE_PREFIX } from '@/utils/shared/thirdwebAuthToken'
 
 export function useThirdwebAuthUser() {
-  const data = useUser<Json, AuthSessionMetadata>()
+  useActiveWalletConnectionStatus()
+
+  const token = Cookies.get(THIRDWEB_AUTH_TOKEN_COOKIE_PREFIX)
+
+  const { userId, address } = useMemo(() => {
+    const decodedToken = token
+      ? jwtDecode<{ ctx?: { userId?: string; address?: string } }>(token)
+      : null
+    const { userId, address } = decodedToken?.ctx ?? {}
+
+    return { userId, address }
+  }, [token])
```

Comment:
I think we merged two possible solution we discussed here, not sure that both are necessary, try one of these:
```suggestion
  const loginStatus = useActiveWalletConnectionStatus()

  const { userId, address } = useMemo(() => {
    const token = Cookies.get(THIRDWEB_AUTH_TOKEN_COOKIE_PREFIX)
    const decodedToken = token
      ? jwtDecode<{ ctx?: { userId?: string; address?: string } }>(token)
      : null
    const { userId, address } = decodedToken?.ctx ?? {}

    return { userId, address }
  }, [loginStatus])
```

**OR**

```suggestion
  const token = Cookies.get(THIRDWEB_AUTH_TOKEN_COOKIE_PREFIX)

  const { userId, address } = useMemo(() => {
    const decodedToken = token
      ? jwtDecode<{ ctx?: { userId?: string; address?: string } }>(token)
      : null
    const { userId, address } = decodedToken?.ctx ?? {}

    return { userId, address }
  }, [token])
```

Reply:
Both of them simplify this code in a different way. If neither work please add a comment to the `useActiveWalletConnectionStatus` call explaining why we aren't using the return value

Reply:
I can't use the `loginStatus` as a dependency on useMemo because React detects that it is not necessary and causes error on the lint verification step. I added a comment with why we are using this `useActiveWalletConnectionStatus` hook without the returned values.